pkg/tarmak/logs/logs.go

```diff
@@ -0,0 +1,337 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package logs
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/hashicorp/go-multierror"
+	"github.com/mitchellh/go-homedir"
+	"github.com/sirupsen/logrus"
+
+	tarmakv1alpha1 "github.com/jetstack/tarmak/pkg/apis/tarmak/v1alpha1"
+	"github.com/jetstack/tarmak/pkg/tarmak/interfaces"
+	"github.com/jetstack/tarmak/pkg/tarmak/utils"
+)
+
+const (
+	timeLayout = "Jan _2 15:04:05"
+)
+
+var (
+	TargetGroups = []string{
+		"bastion",
+		"vault",
+		"etcd",
+		"worker",
+		"control-plane",
+	}
+)
+
+type Logs struct {
+	tarmak interfaces.Tarmak
+	ssh    interfaces.SSH
+	ctx    interfaces.CancellationContext
+	log    *logrus.Entry
+
+	path    string
+	since   string
+	until   string
+	targets []string
+
+	mu       sync.Mutex
+	wg       sync.WaitGroup
+	hosts    []interfaces.Host
+	tmpDir   string
+	tmpFiles map[string]*os.File
```

Comment:
I think you should make sure that this has it's own lock protecting read/writes to the map